core/tests/protractor_utils/editor.js

```diff
@@ -372,28 +387,85 @@ var addParameterChange = function(paramName, paramValue) {
 
 // RULES
 
-// This function selects a rule for the current interaction and enters the
-// entries of the parameterValues array as its parameters; the parameterValues
-// should be specified after the ruleName as additional arguments. For example
-// with interaction 'NumericInput' and rule 'Equals' then there is a single
-// parameter which the given answer is required to equal.
-var _selectRule = function(ruleElement, interactionId, ruleName) {
+var selectRuleInAddResponseModal = function(interactionId, ruleName) {
+  var ruleElement = element(by.css('.protractor-test-add-response-details'));
+  _selectRule(ruleElement, interactionId, ruleName);
+};
+
+var selectRuleParametersInAddResponseModal = function() {
+  var ruleElement = element(by.css('.protractor-test-add-response-details'));
+  var args = [ruleElement];
+  for (var i = 0; i < arguments.length; i++) {
+    args.push(arguments[i]);
+  }
+  _selectRuleParameters.apply(null, args);
+};
+
+// From the corresponding ruleDescription string deduces and returns the types
+// of the parameter objects
+var _getRuleParameterTypes = function(interactionId, ruleName) {
+  var ruleDescription = rules.getDescription(
+    interactions.getInteraction(interactionId).answerObjectType, ruleName);
+
+  var parameterStart = (ruleDescription.indexOf('{{') === -1) ?
+    undefined : ruleDescription.indexOf('{{');
+  var parameterTypes = [];
+  while (parameterStart !== undefined) {
+    var parameterEnd = ruleDescription.indexOf('}}', parameterStart) + 2;
+    parameterTypes.push(
+      ruleDescription.substring(
+        ruleDescription.indexOf('|', parameterStart) + 1, parameterEnd - 2));
+
+    var nextParameterStart =
+      (ruleDescription.indexOf('{{', parameterEnd) === -1) ?
+      undefined : ruleDescription.indexOf('{{', parameterEnd);
+    parameterStart = nextParameterStart;
+  }
+  return parameterTypes;
+};
+
+// This function enters the entries of the parameterValues array as its
```

Comment:
This function sets the parameter values for the given rule. 

Note: The parameter values should be specified as additional arguments after the ruleName. For example, the call

     _selectRuleParameters(ruleElement, 'NumericInput', 'Equals', 24)

will result in a rule that checks if the learner's answer equals 24.